### PR TITLE
User can see list of all games in GAMES tab

### PR DESCRIPTION
**Pull request by karczewskiPiotr**
- Refactored JSON API for Games and Categories to meet standards presented in workshop. 
- Added tests for both normal and JSON API controllers.
- Moved .jsx files to correct folders
- Fixed some bugs in controllers and jbuilder files
- Fixed all rubocop offenses
- Added React component for list of all games.
  As discussed with product owner games are sorted first by rating and then alphabetically.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,9 +1,7 @@
...
-    def index
-        
-    end
+  def show
+    @user = User.find(params[:id])
   end
-  
\ No newline at end of file
+
+  def index; end
```
Comment (hirokiraj) [inline app/controllers/users_controller.rb]: remove unused action

File: app/views/api/v1/games/_game.json.jbuilder [hunk 2]
```diff
@@ -0,0 +1,14 @@
+json.id game.id
+json.title game.title
+json.description game.description
+json.category game.category.name
+json.rating game.rating
```
Comment (hirokiraj) [inline app/views/api/v1/games/_game.json.jbuilder]: extract/call

File: app/views/layouts/application.html.haml [hunk 3]
```diff
@@ -11,12 +11,12 @@
...
-      = link_to "Log out", destroy_user_session_path, method: :delete
-      = link_to "Edit profile", edit_user_registration_path
-    - else
-      = link_to "Log in", new_user_session_path
+    -# = link_to "Binar Games", root_path
+    -# - if user_signed_in?
+    -#   = link_to "Log out", destroy_user_session_path, method: :delete
+    -#   = link_to "Edit profile", edit_user_registration_path
+    -# - else
+    -#   = link_to "Log in", new_user_session_path
```
Comment (hirokiraj) [inline app/views/layouts/application.html.haml]: if it is unused then remove it

File: app/controllers/api/v1/games_controller.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+class Api::V1::GamesController < ApiController
+  rescue_from ActiveRecord::RecordNotFound, with: :handle_record_not_found
```
Comment (hirokiraj) [inline app/controllers/api/v1/games_controller.rb]: consider doing it in more general manner in `ApiController`

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -1,16 +1,17 @@
 class ApplicationController < ActionController::Base
-    protect_from_forgery with: :exception
-    before_action :configure_permitted_parameters, if: :devise_controller?
+  protect_from_forgery with: :exception
+  before_action :configure_permitted_parameters, if: :devise_controller?
 
-     def after_sign_in_path_for(resource)
-          users_path
-     end
+  def after_sign_in_path_for(*)
```
Comment (hirokiraj) [inline app/controllers/application_controller.rb]: A star?

File: app/models/user.rb [hunk 6]
```diff
@@ -8,12 +8,16 @@ class User < ApplicationRecord
   validates :nick, presence: true, uniqueness: true
   has_many :games
 
+  def as_json(options = nil)
```
Comment (hirokiraj) [inline app/models/user.rb]: if it is unused, then we can remove it

File: app/views/api/v1/games/_game.json.jbuilder [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+unless game.pictures.attachments.size.zero?
+  json.pictures(game.pictures) do |picture|
+    json.url Rails.application.routes.url_helpers.rails_blob_path(picture, only_path: true)
+  end
+end
+if game.game_guide.attached?
+  json.guide Rails.application.routes.url_helpers.rails_blob_path(game.game_guide, only_path: true)
+end
+json.user game.user
```
Comment (hirokiraj) [inline app/views/api/v1/games/_game.json.jbuilder]: here, do not use `as_json` but set json keys manually

File: app/javascript/components/games/Rating.jsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    <span className="star-rating">
+      {[1, 2, 3, 4, 5].map(star => (
+        <Star key={star} full={star <= rating || 0} />
+      ))}
+    </span>
+  );
+};
+
+Rating.propTypes = {
+  rating: PropTypes.number || null
```
Comment (Vakrim) [inline app/javascript/components/games/Rating.jsx]: `|| null` is wrong here. `PropTypes.number` already means that it is optional. Use `PropTypes.number.isRequired` in other case

File: app/assets/stylesheets/games/game.scss [hunk 9]
```diff
@@ -0,0 +1,32 @@
+.games-list-wrapper {
+  text-align: center;
+  #games-list {
```
Comment (Vakrim) [inline app/assets/stylesheets/games/game.scss]: avoid using `ids` for styling. Use classes instead

File: app/javascript/components/games/Rating.jsx (hunk 8 above)
Comment (karczewskiPiotr) [reply app/javascript/components/games/Rating.jsx]: Changed it to `PropTypes.number.isRequired` and added a default value of __0__ for `rating` column in database.

**Pull request merged**